extensions/default/src/DicomWebDataSource/utils/clientManager.ts

```diff
@@ -0,0 +1,210 @@
+import { api } from 'dicomweb-client';
+import StaticWadoClient from './StaticWadoClient';
+import dcm4cheeReject from '../dcm4cheeReject';
+import { errorHandler, utils } from '@ohif/core';
+
+export default class clientManager {
+  clients;
+  userAuthenticationService;
+
+  // check google server params
+  private checkGoogleServerParams(params) {
+    const { project, location, dataset, dicomStore } = params;
+    return project && location && dataset && dicomStore;
+  }
+
+  // parsers google server parameter
+  private parseGoogleServerParameter(param) {
+    // remove first slash
+    if (param[0] === '/') {
+      param = param.slice(1);
+    }
+    const tokens = param.split('/');
+    const params = {};
+
+    for (let i = 0; i < Math.floor(tokens.length / 2); i++) {
+      params[tokens[i * 2]] = tokens[i * 2 + 1];
+    }
+
+    if (params['projects']) {
+      params['project'] = params['projects'];
+    }
+    if (params['locations']) {
+      params['location'] = params['locations'];
+    }
+    if (params['datasets']) {
+      params['dataset'] = params['datasets'];
+    }
+    if (params['dicomStores']) {
+      params['dicomStore'] = params['dicomStores'];
+    }
+    return params;
+  }
+
+  // adds a dicomweb server configuration in the clients list
+  private _addConfiguration(givenConfig) {
+    const config = Object.assign({}, givenConfig);
+    config.qidoConfig = {
+      url: config.qidoRoot,
+      staticWado: config.staticWado,
+      singlepart: config.singlepart,
+      headers: this.userAuthenticationService.getAuthorizationHeader(),
+      errorInterceptor: errorHandler.getHTTPErrorHandler(),
+    };
+
+    config.wadoConfig = {
+      url: config.wadoRoot,
+      staticWado: config.staticWado,
+      singlepart: config.singlepart,
+      headers: this.userAuthenticationService.getAuthorizationHeader(),
+      errorInterceptor: errorHandler.getHTTPErrorHandler(),
+    };
+
+    config.qidoDicomWebClient = config.staticWado
+      ? new StaticWadoClient(config.qidoConfig)
+      : new api.DICOMwebClient(config.qidoConfig);
+
+    config.wadoDicomWebClient = config.staticWado
+      ? new StaticWadoClient(config.wadoConfig)
+      : new api.DICOMwebClient(config.wadoConfig);
+
+    config.qidoDicomWebClient.name = config.name;
+    config.wadoDicomWebClient.name = config.name;
+    this.clients.push(config);
+  }
+
+  public addConfiguration(params, query, config) {
+    if (
+      this.checkGoogleServerParams(params) &&
+      config.onConfiguration &&
+      typeof config.onConfiguration === 'function'
+    ) {
+      config = config.onConfiguration(config, {
+        params,
+        query,
+      });
+    }
+    this._addConfiguration(config);
+
+    // handle secondGoogleServer parameter
+    const parameters = [...query.keys()];
+    if (parameters.includes('secondGoogleServer')) {
```

Comment:
I think we should rename this parameter and make it more generic e.g. secondaryServer or maybe a chain of servers if we want to make this scale. Also this should be in a separate mode or extension too.